### PR TITLE
Adding GeLU operator (used in Gpt2)

### PR DESCRIPTION
**Pull request by Narsil**
- Added the operations as found here:

https://github.com/pytorch/pytorch/blob/acdd462b1a070790799ce4623ce8ecc83e197e81/torch/_decomp/decompositions.py#L217
https://github.com/pytorch/pytorch/blob/acdd462b1a070790799ce4623ce8ecc83e197e81/caffe2/operators/gelu_op.cu#L84
https://pytorch.org/docs/stable/generated/torch.nn.GELU.html

- Added both cuda and cpu versions.
- Added a `simplify` helper method, because the results are slightly
  different between cuda and cpu (this is relatively common IMO).

### CODE REVIEW

File: src/tensor_ops/gelu/cpu_kernel.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+use crate::tensor_ops::cpu_kernels::UnaryDerivative;
+use no_std_compat::f32::consts::PI;
```
Comment (coreylowman) [inline src/tensor_ops/gelu/cpu_kernel.rs:2]:
```suggestion
use std::f32::consts::PI;
```
since no_std_compat should overwrite the std namespace

**Review by coreylowman — COMMENTED** [review]
Looks great! Small nits and then good to go. We can use this PR as a model for how to contribute new unary ops 🔥

File: src/tensor_ops/gelu/mod.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        data.iter().map(|x| (x * m).round() / m).collect()
+    }
+
+    #[test]
+    fn test_gelu() {
+        let dev: TestDevice = Default::default();
+        let x = dev.tensor([-2.0, -1.0, 0.0, 1.0, 2.0]);
+        let r = x.trace().gelu();
+        assert_eq!(
+            simplify(&r.array()),
```
[the remark this answers is not in the document]
Comment (coreylowman) [reply src/tensor_ops/gelu/mod.rs:56]: Ahh yeah looks like there's small failure in github actions, I think moving to assert_close should resolve that (probably don't need simplify)

**Comment by Narsil** [comment]
The last failing test is the example I think, were I cannot use `assert_close` (afaik).

Should `assert_close` be opened ? Or maybe relax the assertion in a simple print statement ?

**Comment by Narsil** [comment]
Why was the last test cancelled ??

**Comment by coreylowman** [comment]
> Why was the last test cancelled ??

No idea, but the rest of them passed so looks good to me!

**Comment by coreylowman** [comment]
> The last failing test is the example I think, were I cannot use `assert_close` (afaik).
> 
> Should `assert_close` be opened ? Or maybe relax the assertion in a simple print statement ?

Ah yeah, for doctests you don't need to call any assertions about data, they are more about just showing usage. You can just remove those lines

File: src/tensor_ops/gelu/mod.rs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+/// Examples:
+/// ```rust
+/// # use dfdx::prelude::*;
+/// # let dev: Cpu = Default::default();
+/// let t = dev.tensor([-1.0, 0.0, 1.0, 2.0]);
+/// let r = t.gelu();
+/// // assert_close(
+/// //     &r.array(),
+/// //     &[-0.04540229, -0.158808, 0.0, 0.841192, 1.9545977],
+/// // );
```
Comment (coreylowman) [inline src/tensor_ops/gelu/mod.rs:24]:
```suggestion
```

**Review by coreylowman — APPROVED** [review]
Looks great from my end!

**Comment by Narsil** [comment]
I'll let you merge in then ;) cheers !

**Comment by coreylowman** [comment]
Nice contribution!

**Pull request merged**